gossip/emitter/throttling/throttling_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2025 Sonic Operations Ltd
+// This file is part of the Sonic Client
+//
+// Sonic is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Sonic is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with Sonic. If not, see <http://www.gnu.org/licenses/>.
+
+package throttling
+
+import (
+	"testing"
+
+	"github.com/0xsoniclabs/sonic/inter"
+	"github.com/0xsoniclabs/sonic/opera"
+	"github.com/Fantom-foundation/lachesis-base/inter/idx"
+	"github.com/Fantom-foundation/lachesis-base/inter/pos"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func TestThrottling_SkipEventEmission_DoNotSkipIfBelongingToDominantSet(t *testing.T) {
+
+	tests := map[string]struct {
+		validatorID idx.ValidatorID
+		validators  *pos.Validators
+	}{
+		"validator is equivalent to dominant cutoff": {
+			validatorID: 1,
+			validators:  makeValidators(75, 25),
+		},
+		"validator belongs to dominant set": {
+			validatorID: 2,
+			validators: makeValidators(
+				750, 750, // 75% owned by first two validators
+				125, 125, 125, 125,
+			),
+		},
+		"non-first validator belongs to dominant set": {
+			validatorID: 2,
+			validators: makeValidators(
+				750, 750, // 75% owned by first two validators
+				125, 125, 125, 125,
+			),
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+
+			ctrl := gomock.NewController(t)
+			world := NewMockWorldReader(ctrl)
+			world.EXPECT().GetEpochValidators().Return(test.validators, idx.Epoch(0))
+			world.EXPECT().GetRules().Return(opera.Rules{})
+			world.EXPECT().GetLatestBlockIndex().Return(idx.Block(0)).AnyTimes()
+
+			state := NewThrottlingState(test.validatorID, 0.75, 3, world)
+
+			event := inter.NewMockEventPayloadI(ctrl)
+			event.EXPECT().Transactions().Return(types.Transactions{}).AnyTimes()
+			event.EXPECT().Frame().Return(idx.Frame(1)).AnyTimes()
+
+			skip := state.SkipEventEmission(event)
+			require.False(t, skip)
```

Comment:
Right now, the `state` declares the given event to be skipped for the reasons intended by the test. But to verify this, I had to step through the implementation with a debugger. As discussed offline, it would be good to ensure that events are skipped for the intended reasons -- here and in other tests.

Some ideas how this could be covered:
- renaming the function to "ShouldEmit" and returning an error, explaining why not
- returning an enum-type for the made decision
- have a metric counter that tracks the rejection reasons and check those values before and after